File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.py
class Solution:
    def findPermutationDifference(self, s: str, t: str) -> int:
        count=0
        sc = {char : index for index, char in enumerate(s)}
        tc = {char : index for index, char in enumerate(t)}

        for key in sc:
            count += (abs(sc[key] - tc[key]))

        return count
        

File: 3146-permutation-difference-between-two-strings/3146-permutation-difference-between-two-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var findPermutationDifference = function(s, t) {
  let sc = [...s].reduce((acc, char, index) => {
    acc[char] = index;
    return acc;
  }, {});

  let tc = [...t].reduce((acc, char, index) => {
    acc[char] = index;
    return acc;
  }, {});

  return Object.keys(tc).reduce((sum, char) => {
    return sum + Math.abs(tc[char] - (sc[char] || 0));
  }, 0);
};